src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,201 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useState, useMemo, useEffect } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { PeopleMap } from '../types/PeopleMap';
+import React from 'react';
+
+const getCentury = (year: number): number => {
+  return Math.ceil(year / 100);
+};
 
 export const PeoplePage = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const { slug: selectedPersonSlug } = useParams();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query') ?? '';
+  const sortColumn = searchParams.get('sort') as keyof Person | null;
+  const sortDirection = searchParams.get('order') === 'desc';
+
+  useEffect(() => {
+    getPeople()
+      .then(fetchedPeople => {
+        setPeople(
+          fetchedPeople.map(person => ({
+            ...person,
+            century: person.born ? getCentury(person.born) : undefined,
+          })),
+        );
+      })
+      .catch(() => {
+        setError('Something wrong');
```

Comment:
The error message 'Something wrong' is not very descriptive. Consider providing more detailed error information to help with debugging and improve user experience.